src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs

```diff
@@ -84,20 +88,24 @@ public void JoinThreePaths(string path1, string path2, string path3, string expe
         }
 
         [Theory, MemberData(nameof(TestData_JoinThreePaths))]
-        public unsafe void TryJoinThreePaths(string path1, string path2, string path3, string expected)
+        public void TryJoinThreePaths(string path1, string path2, string path3, string expected)
         {
             char[] output = new char[expected.Length];
 
-            Assert.True(Path.TryJoin(path1, path2, path3, new Span<char>(output), out int written));
+            Assert.True(Path.TryJoin(path1, path2, path3, output, out int written));
             Assert.Equal(expected.Length, written);
+            Assert.Equal(expected, new string(output));
+
             if (expected.Length > 0)
             {
                 Assert.False(Path.TryJoin(path1, path2, path3, Span<char>.Empty, out written));
                 Assert.Equal(0, written);
-                Assert.False(Path.TryJoin(path1, path2, path3, new Span<char>(output, 0, output.Length - 1), out written));
+
+                output = new char[expected.Length - 1];
```

Comment:
Should we assign some dummy data into the char array first and validate it doesn't get reset to an array of 0s?

Reply:
I don't think there would be much value. If someone wants to add it I wouldn't kick it back.